server/server.go

```diff
@@ -211,6 +212,11 @@ func NewServer(opts *Options) (*Server, error) {
 	// Created server's nkey identity.
 	kp, _ := nkeys.CreateServer()
 	pub, _ := kp.PublicKey()
+	serverName := pub
```

Comment:
Mind explaining why? Both Ivan and I saw the use of it being always there.

We see many benefits and thus far cannot understand why you would want it absent by default. Having it there means anyone can rely on it, without it everyone who develops against these structures has to carry complexity and make possibly competing decisions about what the behaviour is and having it omitempty makes it hard to even discover this property exist.

Reply:
What purpose do you want it to serve? Maybe you me and Ivan can jump on a call to discuss.

I think what you are trying to do may have some relevance, but not convinced it should be a default new field for the server info.

Reply:
Lets say you subscribe to server events for monitoring or health tracking etc and you get this:

<details>
 <summary>Server STATZ event</summary>
<pre>
[#49] Received on [$SYS.SERVER.NAND64R34L5D6M4O27JUGJB4AIBIS55IW6VCOG76ZOLYZ3CAYKE6JDDP.STATSZ]: '{
  "server": {
    "host": "0.0.0.0",
    "id": "NAND64R34L5D6M4O27JUGJB4AIBIS55IW6VCOG76ZOLYZ3CAYKE6JDDP",
    "cluster": "c1",
    "ver": "2.1.0",
    "seq": 327313,
    "time": "2019-10-17T12:31:54.954455831Z"
  },
  "statsz": {
    "start": "2019-10-16T16:09:54.057523125Z",
    "mem": 10027008,
    "cores": 2,
    "cpu": 0,
    "connections": 0,
    "total_connections": 0,
    "active_accounts": 2,
    "subscriptions": 25,
    "sent": {
      "msgs": 652184,
      "bytes": 360368849
    },
    "received": {
      "msgs": 327315,
      "bytes": 909733
    },
    "slow_consumers": 100,
    "routes": [
      {
        "rid": 2,
        "sent": {
          "msgs": 324867,
          "bytes": 359456982
        },
        "received": {
          "msgs": 327315,
          "bytes": 909733
        },
        "pending": 0
      }
    ],
    "gateways": [
      {
        "gwid": 6,
        "name": "c2",
        "sent": {
          "msgs": 2,
          "bytes": 2134
        },
        "received": {
          "msgs": 0,
          "bytes": 0
        },
        "inbound_connections": 0
      }
    ]
  }
}'
<pre>
</details>

It's impossible to figure out which machine this relates, like say you have a big cluster or cluster of clusters, there is nothing here that helps an operator or monitoring tool answer the question of "which machine had 100 slow consumers", you cannot go from here to try and debug this.

By default - even with the default of being pk - this is not helpful data. I can't even imagine how you would find out what machine has the PK `NAND64R34L5D6M4O27JUGJB4AIBIS55IW6VCOG76ZOLYZ3CAYKE6JDDP` unless you still have their startup data? You'd have to visit every single machine in every cluster and go do curls against them to find out.

So with the current level of information - and the default behavior you are suggesting we add, or even with the default behavior of just server_name = pk - if someone wanted to build monitoring tools using the events it would not be usable by default unless they set this config option.

At least with `server_name` there and defaulting someone might find a hint that the config item even exists when looking at this data, with it empty and omitempty you would struggle to find how to solve this problem.

Reply:
(`cluster` might at least help you of course narrow it down to some subset of machines, its just gateway name though, doesnt help sites without gateways)

Reply:
I definitely understand your point. And I like the addition, just think it should be opt in vs opt out.

Some other thoughts.

1. host does not have to b 0.0.0.0 and can be assign to $HOST (I understand the localhost issue here).
2. With modern deployment tools like k8s the hostname keeps changing, so in terms of graphing stats about servers no more helpful that server_id, which is unique and is guaranteed to be tied to a single instance of a server.
3. If host is not 0.0.0.0, then then you have the information you need to connect to the machine.

Again, not against additional server_name, just think should be opt in and that our graphing tools should adapt and use it if it's there. This is also for backwards compatibility, users may upgrade monitoring graphs w/o upgrading server.

Reply:
1. no, because hostname isnt related to the ip address software binds to.  Hostname is the kernel name which is why it's in `/proc/sys/kernel/hostname`, in the most simple scenarios these relate but as you said yesterday its very commonly not the case and most people will set to 0.0.0.0. or :: or if they do otherwise it would be some specific IP like a data plane vs management plane, hostname would reflect management plane in that case.  In that case having the host be the identifier would be exactly the wrong thing. 
2. the value of a identifier that match the pod is I can find out what pod is the problem, without it I cannot. Even if its changing its still valuable as its the most current way to find out. Without this information I would have to initiate a series of curls against every pod in every cluster in order to find this piece of obviously missing information - and then the info I will get will be the pod name so I can kubectl the thing to death or exec it or whatever.  Any follow up action from monitoring - even automated - needs a host identifier.
3. No, because this is not what you connect to in kubernetes, you need a name that is meaningful to the management platform.

Reply:
And I'd add to 2 if it changes and we know in stats what is a previous pod name, we could go fetch its logs and see whats going on.  We would have no recourse without it. We would have to fetch all past pod logs and grep for nats server startup where this ID is logged and hope its not rotated away.

Reply:
ok, good points by both of you. You convinced me.

Let's make the change to Name in ServerInfo and then we can approve and merge.